### PR TITLE
Various fixes to the creative inventory

### LINKED ISSUE #583: Various issues with the creative inventory
- Blocks show up in incorrect categories
- Blocks aren't grouped properly
- Blocks aren't ordered correctly

### PR DESCRIPTION
**Pull request by JustTalDevelops**
Resolves #583.

### CODE REVIEW

File: config.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@
   # players to the server list directly.
   ShutdownMessage = "Server closed."
   # AuthEnabled controls whether or not players must be connected to Xbox Live in order to join the server.
-  AuthEnabled = true
+  AuthEnabled = false
```
Comment (DaPigGuy) [inline config.toml:15]: Revert

File: server/world/entity.go [hunk 2]
```diff
@@ -45,7 +45,10 @@ type TickerEntity interface {
 // registered on startup using RegisterEntity to allow loading them in a World.
 type SaveableEntity interface {
 	Entity
-	NBTer
+	// DecodeBlockNBT returns the (new) entity, depending on which of those the NBTer was, with the NBT data decoded into it.
+	DecodeItemNBT(data map[string]any) any
+	// EncodeBlockNBT encodes the entity into a map which can then be encoded as NBT to be written.
+	EncodeItemNBT() map[string]any
```
Comment (DaPigGuy) [inline server/world/entity.go:51]: Mismatch between docs and actual names

File: server/block/banner.go [hunk 3]
```diff
@@ -113,6 +113,34 @@ func (b Banner) DecodeNBT(data map[string]any) any {
 	return b
 }
```
Comment (Sandertv) [inline server/block/banner.go:114]: Can EncodeBlockNBT and DecodeBlockNBT re-use EncodeItemNBT and DecodeItemNBT?

File: server/world/entity.go (hunk 2 above)
Comment (Sandertv) [reply server/world/entity.go:51]: I'm a bit confused as to why a `SaveableEntity` would have `DecodeItemNBT` and `EncodeItemNBT` methods in the first place.

File: server/block/banner.go (hunk 3 above)
Comment (JustTalDevelops) [reply server/block/banner.go:114]: Here, or in general? Here, probably, but not in certain blocks like shucker boxes.

Comment (Sandertv) [reply server/block/banner.go:114]: Yes, I mean here.

File: server/world/entity.go (hunk 2 above)
Comment (JustTalDevelops) [reply server/world/entity.go]: Fixed these.

**Comment by DaPigGuy** [comment]
Issues caused by unregistered direction states have been resolved in the master branch.

Issues with grouping of items should hopefully be resolved with changes to the CreativeContent packet coming in 1.21.60., otherwise, there is a more up-to-date branch with a fix.

Closing

**Pull request closed without merging**